### PR TITLE
VaultFS

### PR DESCRIPTION
**Pull request by BrianHicks**

### CODE REVIEW

File: vault/vaultfs/vaultfs-docker.service [hunk 1]
```diff
@@ -0,0 +1,15 @@
+[Unit]
+Description=Docker volume plugin for vaultfs
+Documentation={{.URL}}
+After=network-online.target
```
Comment (siddharthist) [inline vault/vaultfs/vaultfs-docker.service]: Should this also be after `docker.service`?

Comment (BrianHicks) [inline vault/vaultfs/vaultfs-docker.service]: Yes, probably

**Comment by Dacode45** [comment]
Added Vaultfs support by creating a new service `vaultfs.service` that
mounts `vault` at /var/vault.

File: README.md [hunk 2]
```diff
@@ -26,6 +26,7 @@ for building generic Mantl utilities.
 		- [Vault Packages](#vault-packages)
 			- [vault](#vault)
 			- [vault-mantl](#vault-mantl)
+			- [vault-packages](#vaultfs)
```
Comment (BrianHicks) [inline README.md]: the name doesn't match the header here

File: vault/vault-mantl/spec.yml [hunk 3]
```diff
@@ -15,13 +15,13 @@ depends:
 
 targets:
   - src: "{{.SpecRoot}}/vault-unseal.sh"
-    dest: /usr/local/bin
+    dest: /usr/local/bin/vault-unseal.sh
```
Comment (BrianHicks) [inline vault/vault-mantl/spec.yml]: why did you change these?

File: vault/vaultfs/vaultfs-bootstrap.sh [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+# Constants
+#
+SECURE_TOKEN_PATH="/etc/default/consul_secure.env"
+
+if [ -z "${VAULT_TOKEN}" -a -f ${SECURE_TOKEN_PATH} ]; then
+  source ${SECURE_TOKEN_PATH}
+  VAULT_TOKEN=${SECURE_TOKEN_PATH}
+fi
+
+vaultfs mount --address=$VAULT_ADDRESS -t ${VAULT_TOKEN} ${VAULT_MOUNTPOINT}
```
Comment (BrianHicks) [inline vault/vaultfs/vaultfs-bootstrap.sh]: this will leave the daemon running. Is that what you want?

File: README.md (hunk 2 above)
Comment (Dacode45) [inline README.md]: Fixed.

File: vault/vault-mantl/spec.yml (hunk 3 above)
Comment (Dacode45) [inline vault/vault-mantl/spec.yml]: I got a weird error where the package wouldn't build unless this was set explicitly.

File: vault/vaultfs/vaultfs-bootstrap.sh (hunk 4 above)
Comment (Dacode45) [inline vault/vaultfs/vaultfs-bootstrap.sh]: Fixed. Now runs as a forked process.

**Comment by siddharthist** [comment]
@Dacode45 Looks like something strange has happened to the git history here...

**Comment by Dacode45** [comment]
The commit history looks weird because I did a last minute amend to a commit to add a Readme change. Everything looks okay though.

**Comment by BrianHicks** [comment]
This has drifted very far out of date. I'm happy to help transfer context on a rewrite, but am going to close it for now as I'm not able to be responsible for rewriting/merging it at this point.

**Comment by Theaxiom** [comment]
Thanks @BrianHicks -- if you want to be added to Mantl org, let me know.
